### PR TITLE
chore: add build to deployment composante + nightly flag

### PR DESCRIPTION
**Pull request by louis-bompart**
This PR adds a new kind of deployment: nightly deployments.
Quick-fire descriptions of Nightly Deployments:

They are only for the CDN
They are not for direct consumer uses (i.e. only Hosted Services may use them directly as "stop-gap" measure between official releases.
They are triggered manually
They only target a single "CDN path" per project & major (i.e. no v3.0-nightly, or v3.0.1-nightly, only v3-nightly
This PR concerns v3 deployment. Find the two related PR:

https://github.com/coveo-platform/ui-kit-cd/pull/44 this one for "how the thing will be actually dispatched"
https://github.com/coveo/ui-kit/pull/4650 is the equivalent for v2

### CODE REVIEW

File: .deployment.config.json [hunk 1]
```diff
@@ -48,9 +51,22 @@
         }
       }
     },
+    {
+      "id": "deploy-bueno-nightly-to-s3-version",
+      "s3": {
+        "disabled": $[IS_NIGHTLY],
```
Comment (louis-bompart) [inline .deployment.config.json:57]:
```suggestion
        "disabled": $[IS_NOT_NIGHTLY],
```

**Review by alexprudhomme — APPROVED** [review]
So how would we trigger this ?

**Comment by alexprudhomme** [comment]
> So how would we trigger this ?

Is it a github action in ui-kit-cd ?

**Comment by louis-bompart** [comment]
> > So how would we trigger this ?
> 
> Is it a github action in ui-kit-cd ?

Yes 👍 
Will write brief docs once it works (testing with v2 for now)

**Comment by louis-bompart** [comment]
- [ ] Add missing pieces that has been pushed into `v2` branch

**Pull request merged**